src/commit.c

```diff
@@ -350,7 +350,11 @@ int apk_solver_commit_changeset(struct apk_database *db,
 
 	if (!db->performing_self_upgrade) {
 		if (errors)
-			snprintf(buf, sizeof(buf), "%d errors;", errors);
+			if (errors != 1)
+				snprintf(buf, sizeof(buf), "%d errors;", errors);
+			else
+				snprintf(buf, sizeof(buf), "%d error;", errors);
```

Comment:
I'm not sure if it's actually a good idea in C, but isn't ternary operator better here?

```c
snprintf(buf, errors == 1 ? "%d error;" : "%d errors;", errors);
```